test/models/dx/test_solarradiation_afno.py

```diff
@@ -0,0 +1,220 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+
+import numpy as np
+import pytest
+import torch
+
+from earth2studio.models.dx import SolarRadiationAFNO
+from earth2studio.utils import handshake_dim
+
+
+class PhooAFNOSolarRadiation(torch.nn.Module):
+    """Mock model for testing."""
+
+    def forward(self, x):
+        # x: (batch, variables, lat, lon)
+        # The model expects input shape (batch, variables, lat, lon)
+        # where variables includes the input variables plus sza, sincos_latlon, orography, and landsea_mask
+        # We'll return a tensor of the same shape but with only one variable
+        return torch.zeros_like(x[:, :1, :, :])
+
+
+@pytest.fixture
+def mock_model():
+    """Create a mock model for testing."""
+    model = PhooAFNOSolarRadiation()
+    return model
+
+
+@pytest.mark.parametrize(
+    "x",
+    [
+        torch.randn(1, 1, 1, 31, 721, 1440),
+        torch.randn(2, 1, 1, 31, 721, 1440),
+    ],
+)
+@pytest.mark.parametrize("device", ["cpu", "cuda:0"])
+def test_solarradiation_afno(x, device, mock_model):
+    """Test basic functionality of SolarRadiationAFNO model."""
+    # Create mock tensors for model initialization
+    era5_mean = torch.zeros(31, 1, 1)
+    era5_std = torch.ones(31, 1, 1)
+    ssrd_mean = torch.zeros(1, 1, 1)
+    ssrd_std = torch.ones(1, 1, 1)
+    orography = torch.zeros(1, 1, 721, 1440)
+    landsea_mask = torch.zeros(1, 1, 721, 1440)
+    sincos_latlon = torch.zeros(1, 4, 721, 1440)
+
+    model = SolarRadiationAFNO(
+        core_model=mock_model,
+        freq="6h",
+        era5_mean=era5_mean,
+        era5_std=era5_std,
+        ssrd_mean=ssrd_mean,
+        ssrd_std=ssrd_std,
+        orography=orography,
+        landsea_mask=landsea_mask,
+        sincos_latlon=sincos_latlon,
+    ).to(device)
+    x = x.to(device)
+    coords = OrderedDict(
+        {
+            "batch": np.ones(x.shape[0]),
+            "time": np.array([np.datetime64("2024-01-01")]),
```

Comment:
In the test parameterizations please add a test for multiple times single lead time, single time multiple lead time and then multiple time and lead time to check the model behaves correctly

Reply:
Fixed